pkg/terraform/stages/ibmcloud/stages.go

```diff
@@ -0,0 +1,13 @@
+package ibmcloud
+
+import (
+	"github.com/openshift/installer/pkg/terraform"
+	"github.com/openshift/installer/pkg/terraform/stages"
+)
+
+// PlatformStages are the stages to run to provision the infrastructure in IBM Cloud.
+var PlatformStages = []terraform.Stage{
+	stages.NewStage("ibmcloud", "network"),
+	stages.NewStage("ibmcloud", "bootstrap", stages.WithNormalDestroy()),
+	stages.NewStage("ibmcloud", "master"),
```

Comment:
Is there a particular reason to split these into three stages? I was trying to determine that myself looking at these changes, but it wasn't really clear. In contrast, many of the other platforms are keeping "master" and "network" together.... and rolling them out before "bootstrap". What you have here looks like it works, and there's nothing logically wrong with it as far as I can tell. But, I did see some dependency declarations were deleted with regard to LBs, so I wanted to point this out as a possible alternative. See my PR for libvirt as reference:

https://github.com/openshift/installer/pull/5119/files#diff-d7310ec3465d4577b8f70742e7ecd5a22f463698d1aa807da17041613bcd486eR10

Reply:
Ok thanks for the pointer - no reason particular really, just made logical sense for me to split them this way. Let me refactor this to just 2 stages and see if that plays nicely on IBM Cloud VPC.

Reply:
I have tried to refactor this to 2 stages (cluster, then bootstrap), and attempted a cluster creation, but is not behaving as expected.

From what I am seeing, while the master nodes are polling the MCS LB for its ignition, if there is an update to this LB (adding bootstrap to pool), the master nodes gets into a state where it cannot reach the network. And it seems the only way to recover from this, is rebooting the master nodes. This is an issue with IBM Cloud VPC.

I will try to get some attention on this from the appropriate folks, but in the interim, I propose we stick to the 3 stages (network, bootstrap, master) since we know this has been working pretty well thus far.

Reply:
+1